### PR TITLE
Add Page improvements + minor README addition

### PR DESCRIPTION
**Pull request by coluzziandrea**
# What

- Added contributing link to the general README
- Added minus sign on an expense while editing in the Add Page
- Changed color of values in Add Page
- Adds hint text for description

# Why

It was added in the designs [here](https://www.figma.com/file/6NyY9yqunpbU7HIkbNEAL3/Sossoldi-App?type=design&node-id=0-1&mode=design&t=cbf7CsGaOA27mQqg-0)

image

<details>
  <summary>Screenshots & Recordings</summary>
  
untitled
Screenshot_1709227439
Screenshot_1709227474
Screenshot_1709227501

</details>

### CODE REVIEW

File: lib/pages/add_page/add_page.dart [hunk 1]
```diff
@@ -40,11 +45,80 @@ class _AddPageState extends ConsumerState<AddPage> with Functions {
...
   @override
   Widget build(BuildContext context) {
     final selectedType = ref.watch(transactionTypeProvider);
     final selectedTransaction = ref.watch(selectedTransactionUpdateProvider);
 
+    ref.listen(transactionTypeProvider, (previous, next) {
+      // update the amount text when the transaction type changes
+      _updateAmount();
+    });
+
```
Comment (jackrua) [inline lib/pages/add_page/add_page.dart:121]: If you do it this way when you click on an expense transaction and the "Edit transaction" page opens up you are still missing the minus sign. To add it you have to first change the type of the transaction and then the `_updateAmount` function is called. 

So to be clear on the issue:

1. Click on an expense Transaction
2. The "Editing transaction" page opens up 
3. The value of the transaction is in red but is still missing the minus 

I believe that you could just don't use this callback and simply call the function: 

```suggestion
_updateAmount();
```

But maybe I'm missing something, let me know.

**Review by jackrua — COMMENTED** [review]
Thank you for the contribution! I got some issues here and there but I'm happy to discuss.

File: lib/pages/add_page/add_page.dart (hunk 1 above)
Comment (coluzziandrea) [reply lib/pages/add_page/add_page.dart]: It worked!

File: README.md [hunk 2]
```diff
@@ -69,6 +69,12 @@ The pages that will be implemented are:
 
 If you want to help with this project you are more than welcome! Sossoldi is completely free and open source, the best place to start is from the issue tab and look for something that seems interesting to you. If are unsure on where to start feel free to reach out to us on [Discord](https://discord.com/channels/919139369774891088/1013368203575038012), we will be more than happy to help you find what's best for you!
 
+Also, you can refer to internal docs:
+
+-   [Setup Guide](./docs/setup/setup.md)
+-   [Contribution Guide](./docs/contribution/contribution.md)
+-   [Code of Conduct](./docs/contribution/code_of_conduct.md)
+
```
[the remark this answers is not in the document]
Comment (coluzziandrea) [reply README.md]: Yeah, it makes sense

**Comment by coluzziandrea** [comment]
Thank you for your feedback @jackrua, I've added the changes you suggested!!!

**Comment by mikev-cw** [comment]
Super clear PR, and very good point addressed by @jackrua. What a pleasure, thank you both!!
Looks good to me, if also @theperu is ok with the documentation part I think we can safely merge

**Review by theperu — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**